src/cache_test.ts

```diff
@@ -143,11 +143,11 @@ describe('ListWatchCache', () => {
         expect(pathOut).to.equal('/some/path');
         expect(cache.list()).to.deep.equal(list);
 
-        expect(cache.get('name1')).to.equal(list[0]);
-        expect(cache.get('name2')).to.equal(list[1]);
+        expect(cache.get('name1', 'default')).to.equal(list[0]);
+        expect(cache.get('name2', 'default')).to.equal(list[1]);
```

Comment:
I changed the test here to take the `default` namespace. The old logic didn't make much sense and it seemed like it was supposed to return `undefined` if the name and namespace did not match. From a consumer perspective I would expect the namespace to only be optional if the resource is global like a `namespace` or `pvc`. Let me know I should restore the old behavior of finding the first object with a matching name regardless of what namespace its in if no namespace is provided.

Reply:
In the case of this test, the objects are global, they are `V1Namespace` so I think this test shouldn't change.

Reply:
Ah I see this test incorrectly puts `namespace` in the mocked responses so the namespace objects have namespaces. I'll split this into two tests one for namespace objects and one for unnamespaced objects.